src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -428,4 +462,79 @@ public void register() {
 			DispenserBlock.registerBehavior(item, (source, stack) -> condition.test(source, stack) ? behavior.dispense(source, stack) : oldBehavior.dispense(source, stack));
 		}
 	}
+
+	/**
+	 * When an instance of this class is registered using
+	 * {@link DataUtil#registerNoteBlockInstrument(CustomNoteBlockInstrument)}, note blocks will play a custom sound
+	 * if a blockstate predicate for the position under the note block passes. See constructor for details.
+	 *
+	 * <p>If multiple mods add new instruments the predicates may overlap, which is what
+	 * {@code modIdComparator} is intended to solve.</p>
+	 *
+	 * @author abigailfails
+	 */
+	public static class CustomNoteBlockInstrument implements Comparable<CustomNoteBlockInstrument> {
+		protected final String modId;
+		protected final Comparator<String> modIdComparator;
+		protected final Predicate<BlockState> condition;
+		public final SoundEvent sound;
```

Comment:
What I meant before is this field should be `private`.

Reply:
Done.